src/redux/features/votes/votesAsyncThunks.ts

```diff
@@ -0,0 +1,43 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { Response2 } from 'src/services/HttpService';
+import VotesService from 'src/services/VotesService';
+import strings from 'src/strings';
+import {
+  DeleteProjectVotesRequestPayload,
+  DeleteProjectVotesResponsePayload,
+  UpsertProjectVotesRequestPayload,
+  UpsertProjectVotesResponsePayload,
+} from 'src/types/Votes';
+
+export const requestProjectVotesUpdate = createAsyncThunk(
+  'votes/update',
+  async (request: { projectId: number; payload: UpsertProjectVotesRequestPayload }, { rejectWithValue }) => {
+    const response: Response2<UpsertProjectVotesResponsePayload> = await VotesService.setProjectVotes(
+      request.projectId,
+      request.payload
+    );
+
+    if (response !== null && response.requestSucceeded) {
+      return response.data;
+    }
+
+    return rejectWithValue(strings.GENERIC_ERROR);
+  }
+);
+
+export const requestProjectVotesDelete = createAsyncThunk(
+  'votes/delete',
+  async (request: { projectId: number; payload: DeleteProjectVotesRequestPayload }, { rejectWithValue }) => {
+    const response: Response2<DeleteProjectVotesResponsePayload> = await VotesService.deleteProjectVotes(
+      request.projectId,
+      request.payload
+    );
+
+    if (response !== null && response.requestSucceeded) {
+      return response.data;
+    }
+
+    return rejectWithValue(strings.GENERIC_ERROR);
+  }
+);
```

Comment:
do we need a requestListProjectVotes?

Reply:
Ooh you're right, will add 👍

Reply:
Thanks Josh! Just fyi I have branched off yours for the page view.

Reply:
Ok thanks for the heads up, hopefully the changes I'm making haven't been too disruptive to your work

Reply:
The changes were helpful, I needed them :)

Reply:
I added a `requestProjectVotesGet` async thunk, does we need a separate `requestListProjectVotes` thunk?

Reply:
I realize the list thunk sorts data by locale, but I think we still needed `requestProjectVotesGet` to return the entire response.

Reply:
No we don't need a list project votes. We won't be showing votes across projects.
The only way to show votes is by project. When the user visits a project's scores, there's a link to view the project's votes.

Reply:
My original comment/question should have just said 'requestGetProjectVotes' to avoid confusion. Was mostly referring to listing votes within a project, but list also sounds like listing all projects votes. What you have right now is good 👍